server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -950,14 +950,27 @@ protected void finalize(final List<SnapshotFiles> snapshots,
 
             final BlobStoreIndexShardSnapshots updatedSnapshots = new BlobStoreIndexShardSnapshots(snapshots);
             try {
+                // Delete temporary index files
```

Comment:
can we just add a comment why we need to do this first?

Reply:
sure